x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/types.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export enum OpenModal {
+  None,
+  ConfirmModifyModal,
+  ConfirmResetModal,
+  ConfirmSaveModal,
+}
+export interface ServerFieldResultSetting {
+  raw?:
+    | {
+        size?: number;
+      }
+    | boolean;
+  snippet?:
+    | {
+        size?: number;
+        fallback?: boolean;
+      }
+    | boolean;
+}
+
+export type ServerFieldResultSettingObject = Record<string, ServerFieldResultSetting>;
+
+export interface FieldResultSetting {
+  raw: boolean;
+  rawSize?: number;
+  snippet: boolean;
+  snippetSize?: number;
+  snippetFallback: boolean;
+}
```

Comment:
[NOT a change request, just thinking out loud] Is there really a strong advantage to flattening out the result setting obj vs. just maintaining the nested raw vs snippet data structure? Honestly, half the difficulty of parsing out this logic file is constantly trying to context switch between the "server" data and "client-side" data and I wonder if it's really necessary or helpful.

Reply:
I don't know whether there is an advantage or not. I think if this happens it should really only happen at the point of interaction with the server.